lib/droplet_kit.rb

```diff
@@ -62,10 +62,27 @@ module DropletKit
 
   # Utils
   autoload :PaginatedResource, 'droplet_kit/paginated_resource'
+  autoload :ErrorHandlingResourcable, 'droplet_kit/error_handling_resourcable'
 
   # Errors
   autoload :ErrorMapping, 'droplet_kit/mappings/error_mapping'
   Error = Class.new(StandardError)
   FailedCreate = Class.new(DropletKit::Error)
   FailedUpdate = Class.new(DropletKit::Error)
+
+  class RateLimitReached < DropletKit::Error
+    attr_accessor :limit, :remaining, :reset
+
+    def limit
+      @limit.to_i if @limit
+    end
+
+    def remaining
+      @remaining.to_i if @remaining
+    end
+
+    def reset_at
```

Comment:
Whoops, fixed - I had some vague memory of where the client converted timestamp strings to objects, but I couldn't find any after looking, so removed this.